### PR TITLE
xtask: Enable support for rv64

### PR DESCRIPTION
**Pull request by furquan-rivos**
This change enables support for building rv64. PLIC and MTIMER are currently marked as RV32 only. These can be enabled in the future if needed.

### CODE REVIEW

File: build/xtask/src/config.rs [hunk 1]
```diff
@@ -344,6 +344,10 @@ impl Config {
             "riscv32imc-unknown-none-elf" | "riscv32imac-unknown-none-elf" => {
                 MpuAlignment::Chunk(16)
             }
+            // Qemu on RV64 assumes entire page(4KiB) has the same access permissions
```
Comment (adurbin-rivos) [inline build/xtask/src/config.rs:347]: I think we may, in the future, want to break things out based on a configuration. e.g. s-mode/u-mode without pmps, m-mode/u-mode with pmps, etc. That would then allow a better represenation of alignment requirements. And it's not really a QEMU requirement.

**Review by adurbin-rivos — APPROVED** [review]

File: build/xtask/src/dist.rs [hunk 2]
```diff
@@ -883,15 +898,11 @@ fn update_image_header(
...
 ) -> Result<bool> {
-    use goblin::container::Container;
     use zerocopy::AsBytes;
 
     let mut file_image = std::fs::read(input)?;
     let elf = goblin::elf::Elf::parse(&file_image)?;
 
-    if elf.header.container()? != Container::Little {
-        bail!("where did you get a big-endian image?");
-    }
```
Comment (FawazTirmizi) [inline build/xtask/src/dist.rs:894]: Why was this check disabled, both here and in `load_elf()`?

Comment (zachreizner) [reply build/xtask/src/dist.rs:894]: It refers to the whether the elf is 64-bit or not: https://docs.rs/goblin/latest/goblin/container/enum.Container.html

**Review by zachreizner — APPROVED** [review]

**Review by FawazTirmizi — APPROVED** [review]

File: build/xtask/src/config.rs (hunk 1 above)
Comment (furquan-rivos) [reply build/xtask/src/config.rs:347]: Agreed. We will need some configs around this to decide how to handle the different requirements for different cases.

With respect to qemu, there is a patch that fixes the current behavior: https://lists.nongnu.org/archive/html/qemu-devel/2022-09/msg01445.html

**Pull request merged**